src/proxy-ui-api/frontend/src/views/Settings/BackupAndRestore/RestoreBackupButton.vue

```diff
@@ -0,0 +1,77 @@
+<template>
+  <small-button
+    v-if="canBackup"
+    :min_width="50"
+    class="xrd-table-button"
+    data-test="backup-restore"
+    @click="showConfirmation = true"
+    >{{ $t('action.restore') }}
+    <confirm-dialog
+      :dialog="showConfirmation"
+      :loading="restoring"
+      title="backup.action.restore.dialog.title"
+      text="backup.action.restore.dialog.confirmation"
+      :data="{ file: backup.filename }"
+      @cancel="showConfirmation = false"
+      @accept="restoreBackup"
+    />
+  </small-button>
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+import { Prop } from 'vue/types/options';
+import { Backup } from '@/types';
+import * as api from '@/util/api';
+import SmallButton from '@/components/ui/SmallButton.vue';
+import ConfirmDialog from '@/components/ui/ConfirmDialog.vue';
+export default Vue.extend({
+  name: 'RestoreBackupButton',
+  components: {
+    SmallButton,
+    ConfirmDialog,
+  },
+  props: {
+    canBackup: {
+      type: Boolean,
+      default: false,
+      required: true,
+    },
+    backup: {
+      type: Object as Prop<Backup>,
+      required: true,
+    },
+  },
+  data() {
+    return {
+      showConfirmation: false,
```

Comment:
Variables in data-block can be typed. At least in the case of objects it can be usefull and it works at least partially with templates and other methods.

Reply:
Added typings, good idea to always add them.